lib/ReactViews/HelpScreens/HelpPanel.jsx

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+import React from 'react';
+import ObserveModelMixin from '../ObserveModelMixin';
+import defined from 'terriajs-cesium/Source/Core/defined';
+import classNames from 'classnames';
+import MenuPanel from '../StandardUserInterface/customizable/MenuPanel.jsx';
+
+import Styles from './help-panel.scss';
+import DropdownStyles from '../Map/Panels/panel.scss';
+import helpIcon from '../../../wwwroot/images/icons/help.svg';
+
+const HelpPanel = React.createClass({
+    mixins: [ObserveModelMixin],
+
+    propTypes: {
+        viewState: React.PropTypes.object.isRequired,
+        helpSequences: React.PropTypes.object
+    },
+
+    componentWillMount() {
+        const that = this;
+        setInterval(function() {
+            if (!defined(that.state.screens)) {
+                return;
+            }
+            if (that.props.helpSequences.cancel) {
+                // Has been cancelled from somewhere else. Abort!
+                that.cancel();
+            }
+            const i = that.state.index;
+            const currentScreen = that.state.screens[i];
+            if (defined(that.state.previousRectangle) && that.state.previousRectangle === that.state.currentRectangle) {
```

Comment:
Why not `updateCurrentRectangle` before checking to see if the rectangle changed?  Seems like it would simplify the logic and avoid the help panel being a "frame behind".